src/Terminal/Parser.ts

```diff
@@ -33,98 +42,62 @@ export function ParseCommands(commands: string): string[] {
 }
 
 export function ParseCommand(command: string): (string | number | boolean)[] {
-  // This will be used to keep track of whether we're in a quote. This is for situations
-  // like the alias command:
-  //      alias run="run NUKE.exe"
-  // We want the run="run NUKE.exe" to be parsed as a single command, so this flag
-  // will keep track of whether we have a quote in
-  let inQuote = ``;
-
-  // Returns an array with the command and its arguments in each index
-  // Properly handles quotation marks (e.g. `run foo.script "the sun"` will return [run, foo.script, the sun])
+  let idx = 0;
   const args = [];
-  let start = 0,
-    i = 0;
-  let prevChar = ""; // Previous character
-  while (i < command.length) {
-    let escaped = false; // Check for escaped quotation marks
-    if (i >= 1) {
-      prevChar = command.charAt(i - 1);
-      if (prevChar === "\\") {
-        escaped = true;
-      }
-    }
 
-    const c = command.charAt(i);
-    if (c === KEY.DOUBLE_QUOTE) {
-      // Double quotes
-      if (!escaped && prevChar === KEY.SPACE) {
-        const endQuote = command.indexOf(KEY.DOUBLE_QUOTE, i + 1);
-        if (endQuote !== -1 && (endQuote === command.length - 1 || command.charAt(endQuote + 1) === KEY.SPACE)) {
-          args.push(command.substr(i + 1, endQuote - i - 1));
-          if (endQuote === command.length - 1) {
-            start = i = endQuote + 1;
-          } else {
-            start = i = endQuote + 2; // Skip the space
-          }
-          continue;
-        }
-      } else if (inQuote === ``) {
-        inQuote = KEY.DOUBLE_QUOTE;
-      } else if (inQuote === KEY.DOUBLE_QUOTE) {
-        inQuote = ``;
-      }
-    } else if (c === KEY.QUOTE) {
-      // Single quotes, same thing as above
-      if (!escaped && prevChar === KEY.SPACE) {
-        const endQuote = command.indexOf(KEY.QUOTE, i + 1);
-        if (endQuote !== -1 && (endQuote === command.length - 1 || command.charAt(endQuote + 1) === KEY.SPACE)) {
-          args.push(command.substr(i + 1, endQuote - i - 1));
-          if (endQuote === command.length - 1) {
-            start = i = endQuote + 1;
-          } else {
-            start = i = endQuote + 2; // Skip the space
-          }
-          continue;
-        }
-      } else if (inQuote === ``) {
-        inQuote = KEY.QUOTE;
-      } else if (inQuote === KEY.QUOTE) {
-        inQuote = ``;
-      }
-    } else if (c === KEY.SPACE && inQuote === ``) {
-      const arg = command.substr(start, i - start);
+  // Track depth of quoted strings, e.g.: "the're 'going away' rather 'quickly \"and awkwardly\"'" should be parsed as a single string
```

Comment:
That's going to be weird. Generally speaking you don't match quotes inside quotes, otherwise `"they're"` has an unbalanced quote.

Reply:
Hmm, I guess I did get a bit overly fancy there. I can cut it down to simple matching of the start quote, though the current processing defines valid start patterns of ` "` and `="` (and also with single quotes), so it currently doesn't break on an apostrophe if it's unescaped. Thinking on it further, there are a few other edge cases which would potentially break the current codeso I'll go ahead and reduce it to matching the start quote of the string.